code/game/machinery/hologram.dm

```diff
@@ -86,6 +86,11 @@ For the other part of the code, check silicon say.dm. Particularly robot talk.*/
 		rendered_message = "<i><span class='[speech.render_wrapper_classes()]'>Holopad received, <span class='message'>[rendered_message]</span></span></i>"
 		master.show_message(rendered_message, 2)
 
+/obj/machinery/hologram/holopad/on_see(var/message, var/blind_message, var/drugged_message, var/blind_drugged_message, atom/A)
+	if(master && master.eyeobj && master.eyeobj.high_res) //prevent duplicate visible messages 
```

Comment:
Oh, are you referring to the case where the cameras are high res but the holopad is somehow off camera? Yeah, that would cause problems. 
I'm pretty sure the holopad speech/visual message receiving is independent of the cameranet, but actual sight is still handled by the eye I think.
Need to double check for clarity